### ISSUE TITLE
FinalLocalVariable false-positive in SWITCH

### ISSUE THREAD

**Issue by Vladlis** [checkstyle/checkstyle#2405]
Input:
```java
class Test {
    public void method(int i) {
        switch (i) {
            case 1:
                int foo = 1;     // line 5
                break;
            case 2:
                foo = 2;         // line 8
                break;
            default:
        }
    }
}
```
Config:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="TreeWalker">
		<module name="FinalLocalVariable"/>
    </module>
</module>
```
Result:
```bash
>java -jar checkstyle-6.11.2-all.jar -c my_check.xml Test.java
Starting audit...
C:\tmp\Test.java:5:21: warning: Variable 'foo' should be declared final.
Audit done.
```
If foo is declared <code>final</code> on the line 5, then a compilation error appears on the line 8.

**Comment by romani** [comment]
Checkstyle works only on compilable sources , we could be launched on noncompilbale sources but behaviour could be unpredictable.

@Vladlis , Please give compilable input file , or close the issue.

**Comment by Vladlis** [comment]
@romani, the one I've provided is compilable for sure.
Here's one more example: https://github.com/Vladlis/sevntu.checkstyle/commit/d9beb6838cc729685d4c8037c02f115cee85a4c5#diff-1e91aff95af191f32c4f71b5cdc59ad2R125

**Comment by Vladlis** [comment]
Suppression can be removed after the fix:
https://github.com/sevntu-checkstyle/sevntu.checkstyle/commit/cf877fba57ab81a75513bfea26c433106f56c182

**Comment by romani** [comment]
fix is merged
